Add notes for Data plane vs Control Plane and relative setup steps for each (#11745)
DataDog/integrations-core: istio/README.md

### Configuration

> Edit the `istio.d/conf.yaml` file (in the `conf.d/` folder at the root of your [Agent's configuration directory][7]) to connect to Istio. See the [sample istio.d/conf.yaml][8] for all available configuration options.

Reviewer: I think we should leave this statement in, since all checks use this wording

Author: Added back in 👍

Reviewer: Thanks!

> The Istio integration has two key components for how it collects the prometheus formatted Istio metrics. This corresponds to the [Istio architecture][23] split between its data plane (the `istio-proxy` sidecar containers) and the control plane (the `istiod` service managing the proxies). These are both ran as `istio` Agent checks, however have different responsibilities and configuration methods.

Reviewer 2: ```suggestion
There are two key components involved in setting up the Istio integration to collect the Prometheus-formatted Istio metrics. Aligning with the [Istio architecture][23], there is the **data plane** (the `istio-proxy` sidecar containers) and the **control plane** (the `istiod` service managing the proxies). These are both run as `istio` Agent checks, but have different responsibilities, and they are  configured separately, as described below.
```

It's not often I recommend starting sentences with "there is", lol, but this is one of those times. I think it makes what you're saying a little clearer. Look okay to you?

> There are two key components involved in setting up the Istio integration to collect the Prometheus-formatted Istio metrics. Aligning with the [Istio architecture][23], there is the **data plane** (the `istio-proxy` sidecar containers) and the **control plane** (the `istiod` service managing the proxies). These are both run as `istio` Agent checks, but have different responsibilities, and they are  configured separately, as described below.

Reviewer: Can we also continue mentioning this is for Istio `v1.5+`?

Author: Do we still need to make this distinction? Istio 1.4 and under received their end-of-life as of June 2020. I don't have numbers to back this, but customers should be on 1.5+ by now.  

That being said I did note this in the `Control plane configuration` section. Is that sufficient? If not I can add something like that here.

Reviewer: I think it's worth it to keep it in considering 2020 was not too long ago.

Author: Added 1.5+ note back in

##### Data plane configuration

> To monitor the Istio data plane the Agent comes with an [`istio.d/auto_conf.yaml`][9] file to automatically setup the monitoring for each of the `istio-proxy` sidecar containers. The Agent initializes this check for each sidecar container that it discovers automatically. This portion reports the `istio.mesh.*` metrics with respect to the data exposed by each of these sidecar containers.

Reviewer 2: ```suggestion
To monitor the Istio data plane, the Agent includes an [`istio.d/auto_conf.yaml`][9] file to automatically set up the monitoring for each of the `istio-proxy` sidecar containers. The Agent initializes this check for each sidecar container that it detects automatically. This configuration enables the reporting of `istio.mesh.*` metrics for the data exposed by each of these sidecar containers.
```

> To customize this portion of the integration create an equivalent [configuration file][24] for Istio. This should have the `ad_identifiers` and `istio_mesh_endpoint` set appropriately to setup the integration when an `istio-proxy` sidecar container is discovered. Refer to the existing [`istio.d/auto_conf.yaml`][9] as well as the [example configuration file][8] for all available configuration options. When customizing, set the `exclude_labels` to the following configuration:

Reviewer 2: ```suggestion
To customize the data plane portion of the integration, create an equivalent [configuration file][24] for Istio. Set `ad_identifiers` and `istio_mesh_endpoint` appropriately to set up the integration when an `istio-proxy` sidecar container is discovered. Refer to the provided [`istio.d/auto_conf.yaml`][9] and the [example configuration file][8] for all available configuration options. When you customize, set the `exclude_labels` to the following configuration:
```

##### Data plane configuration
To monitor the Istio data plane, the Agent includes an [`istio.d/auto_conf.yaml`][9] file to automatically set up the monitoring for each of the `istio-proxy` sidecar containers. The Agent initializes this check for each sidecar container that it detects automatically. This configuration enables the reporting of `istio.mesh.*` metrics for the data exposed by each of these sidecar containers.

To customize the data plane portion of the integration, create an equivalent [configuration file][24] for Istio. Set `ad_identifiers` and `istio_mesh_endpoint` appropriately to set up the integration when an `istio-proxy` sidecar container is discovered. Refer to the provided [`istio.d/auto_conf.yaml`][9] and the [example configuration file][8] for all available configuration options. When you customize, set the `exclude_labels` to the following configuration:

> ```yaml

Reviewer: One more thing while you are updating the readme: can you make a note to enable `use_openmetrics: true` when customizing as well?

Author: Included this as well 👍

      - request_protocol
      - connection_security_policy
```

> These excluded labels are set in the existing `istio.d/auto_conf.yaml` file.

Reviewer: ```suggestion
```

##### Control plane configuration

> To monitor the Istio control plane and report the remaining `mixer`, `galley`, `pilot`, and `citadel` metrics the Agent needs to be configured to monitor the `istiod` deployment. In Istio `v1.5+`, apply the follow Autodiscovery Annotations as pod annotations for the deployment `istiod` in the `istio-system` namespace:

Reviewer 2: ```suggestion
To monitor the Istio control plane and report the `mixer`, `galley`, `pilot`, and `citadel` metrics, you must configure the Agent to monitor the `istiod` deployment. In Istio `v1.5+`, apply the following Autodiscovery Annotations as pod annotations for the deployment `istiod` in the `istio-system` namespace:
```

```yaml
ad.datadoghq.com/discovery.check_names: '["istio"]'
ad.datadoghq.com/discovery.init_configs: '[{}]'
ad.datadoghq.com/discovery.instances: |
     [
       {
         "istiod_endpoint": "http://%%host%%:15014/metrics",

>          "use_openmetrics": "true"

Reviewer: Can we include `exclude_labels` here as well?

Author: The existing `exclude_labels` that we reference are specific to the data plane portion of this integration. Those labels don't show up in the control plane istiod output so they shouldn't be necessary for this specific portion of the integration. 

But if there any specific `exclude_labels` that you wanted for these metrics I can get those added into the docs:
- https://github.com/DataDog/integrations-core/blob/master/istio/tests/fixtures/1.5/istiod.txt

Reviewer: Ah that's true, my mistake.

       }
     ]
```
The method for applying these annotations varies depending on the [Istio deployment strategy (Istioctl, Helm, Operator)][22] used. Consult the Istio docs for the proper method to apply these pod annotations.

> These annotations reference `discovery` as the `<CONTAINER_IDENTIFIER>` to match the default container name of the pods for the `istiod` deployment. If your container name is different adjust accordingly.

Reviewer 2: ```suggestion
These annotations reference `discovery` as the `<CONTAINER_IDENTIFIER>` to match the default container name of the pods for the `istiod` deployment. If your container name is different, adjust accordingly.
```
